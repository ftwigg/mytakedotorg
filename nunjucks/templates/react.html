{% extends "reactCore.html" %}

{% block bundle %}
	<!-- Main -->
	<script src="/app.bundle.js"></script>
{% endblock %}
