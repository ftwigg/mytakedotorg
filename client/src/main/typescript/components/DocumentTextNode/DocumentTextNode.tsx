import * as React from "react";
import * as ReactDOM from "react-dom";
import { FoundationNode } from "../../utils/functions";

interface DataAttributes {
  "data-char-offset": number;
}

interface DocumentTextNodeProps {
  documentNode: FoundationNode;
}

interface DocumentTextNodeState {}

class DocumentTextNode extends React.Component<
  DocumentTextNodeProps,
  DocumentTextNodeState
> {
  constructor(props: DocumentTextNodeProps) {
    super(props);
  }
  render() {
    const { documentNode } = this.props;

    let attributes: DataAttributes = {
      "data-char-offset": documentNode.props.offset
    };

    switch (documentNode.original.component) {
      const content = splitNodes(documentNode);
      case "h2":
        return (
          <h2 {...attributes}>
            {content}
          </h2>
        );
      case "h3":
        return (
          <h3 {...attributes}>
            {content}
          </h3>
        );
      case "p":
        return (
          <p {...attributes}>
            {content}
          </p>
        );
      default:
        // Unknown element, silently throw error here
        return <div />;
    }
  }
}

export default DocumentTextNode;
