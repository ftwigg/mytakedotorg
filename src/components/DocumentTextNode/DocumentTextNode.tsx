import * as React from "react";
import * as ReactDOM from "react-dom";
import { FoundationNode } from "../../utils/functions";

interface DocumentTextNodeProps {
  documentNode: FoundationNode;
}

interface DocumentTextNodeState {}

class DocumentTextNode extends React.Component<
  DocumentTextNodeProps,
  DocumentTextNodeState
> {
  constructor(props: DocumentTextNodeProps) {
    super(props);
  }
  render() {
    const { documentNode } = this.props;
    switch (documentNode.component) {
      case "h2":
        return (
          <h2 data-offset={documentNode.props.offset}>
            {documentNode.innerHTML}
          </h2>
        );
      case "h3":
        return (
          <h3 data-offset={documentNode.props.offset}>
            {documentNode.innerHTML}
          </h3>
        );
      case "p":
        return (
          <p data-offset={documentNode.props.offset}>
            {documentNode.innerHTML}
          </p>
        );
      default:
        // Unknown element, silently throw error here
        return <div />;
    }
  }
}

export default DocumentTextNode;
