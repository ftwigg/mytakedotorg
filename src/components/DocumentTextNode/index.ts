import DocumentTextNode from "./DocumentTextNode";

export default DocumentTextNode;
