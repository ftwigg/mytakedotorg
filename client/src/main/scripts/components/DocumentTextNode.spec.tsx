import * as React from "react";
import * as renderer from "react-test-renderer";
import {} from "jest";
import DocumentTextNode from "./DocumentTextNode";

const documentNode = {
  component: "p",
  innerHTML: [
    "Section 1. Neither slavery nor involuntary servitude, except as a punishment for crime whereof the party shall have been duly convicted, shall exist within the United States, or any place subject to their jurisdiction."
  ],
  offset: 0
};

test("DocumentTextNode", () => {
  const tree = renderer
    .create(<DocumentTextNode documentNode={documentNode} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
