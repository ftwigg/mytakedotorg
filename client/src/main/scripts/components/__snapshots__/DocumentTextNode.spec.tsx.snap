// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentTextNode 1`] = `
<p
  data-char-offset={0}
>
  Section 1. Neither slavery nor involuntary servitude, except as a punishment for crime whereof the party shall have been duly convicted, shall exist within the United States, or any place subject to their jurisdiction.
</p>
`;
